main.go

```diff
@@ -162,13 +166,48 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "PhSchedule")
 		os.Exit(1)
 	}
-	if err = (&controllers.PhDeploymentReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("PhDeployment"),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "PhDeployment")
-		os.Exit(1)
+
+	// get the ingress from the config which is from the helm value.
```

Comment:
1. Please put the ingress relative data in a single struct. And unmarshal against this struct.
2. Put the unmarshal logic in `if modelDeployment` statement

Reply:
fixed